pkg/segment/query/processor/sortcommand.go

```diff
@@ -162,10 +176,10 @@ func getRank(CValEnc *segutils.CValueEnclosure) dTypeRank {
 }
 
 // Returns A comparison B
-func compareValues(valueA, valueB *segutils.CValueEnclosure, asc bool) compare {
+func compareValues(valueA, valueB *segutils.CValueEnclosure, asc bool, op string) compare {
 	var result compare
-	rankA := getRank(valueA)
-	rankB := getRank(valueB)
+	rankA := getRank(valueA, op)
```

Comment:
the `op` is specified during query time (or defaulted to some value). This `getRank` is getting called 2 x numRecords. The switch/case statement inside it also gets executed numRecords times.  When a explicit rank like str/num is passed in the query, then we should not waste time to find out the rank per record.